nemo/collections/nlp/models/language_modeling/megatron_base_model.py

```diff
@@ -148,6 +149,12 @@ def __init__(self, cfg: DictConfig, trainer: Trainer, no_lm_init=True):
         self.model_parallel_config: ModelParallelConfig = self.build_model_parallel_config()
 
         self.with_distributed_adam = cfg.optim.get('name') == 'distributed_fused_adam'
+        self.use_mcore_dist_optim = cfg.optim.get('use_mcore_dist_optim', False)
```

Comment:
I think it is better to set `self.use_mcore_dist_optim` using `optim.name` instead of adding another arg?

```
self.use_mcore_dist_optim = cfg.optim.get('name') == 'mcore_distributed_fused_adam'
self.with_distributed_adam = cfg.optim.get('name') == 'distributed_fused_adam' or self.use_mcore_dist_optim
```

Reply:
Sure for the rest of naming I will change it accordingly and make it the same as the apex distributed optimizer since apex is going to be deprecated.